### PR TITLE
Deprecate old world and organism functions

### PR DESCRIPTION
**Pull request by cliff-bohm**
removed AbstractWorld function evaluateSolo (and updated Xor world) AND removed genome and brain from Organism. These features had been deprecated but remained in the code for backward compatibility.

### CODE REVIEW

**Comment by cgnitash** [comment]
TestWorld also declares and uses EvaluateSolo(). That needs to be purged as well.

**Comment by cgnitash** [comment]
If I understand this correctly, brain and genome used to be single objects, and now they have to be accessed through a map of strings->brain/genomes. Correct?

**Comment by cgnitash** [comment]
Also, why have you commented out the old lines instead of deleting them?

**Comment by cliff-bohm** [comment]
a) use of evalSolo function. We do not need to purge evaluateSolo functions. If the world designer wishes to write their world with this function, that's fine. We simply do not provide it in the abstract class.

b) a while, organisms had one brain and one genome. A change was made to allow for multiple brains and multiple genomes. At that time, there was a request to maintain brain and genome for backward compatibility. brain was set to brains["root::"] if there was a root brain and genome was set to genomes["root::"], again if there was a root genome. Accessing brains["root::"] should have been identical to accessing brain before this change. If this was not the case, then there is/was a bug.

c) There is no reason to keep the commented out lines, they have been deleted and the changes pushed.

File: World/AbstractWorld.h [hunk 1]
```diff
@@ -50,10 +50,4 @@ class AbstractWorld {
               << std::endl;
```
Comment (cgnitash) [inline World/AbstractWorld.h]: evaluate should be made pure virtual, so the compiler can provide this error.

Comment (cliff-bohm) [reply World/AbstractWorld.h]: Yup. change made.

**Comment by cliff-bohm** [comment]
I agree with the changes. 
AbstractWorld::evaluate is not pure virtual
TestWorld::evaluateSolo is not virtual.

**Pull request merged**